packages/grafana-ui/src/components/Toggletip/Toggletip.tsx

```diff
@@ -0,0 +1,287 @@
+  footer?: JSX.Element | string;
+  /** The UI control users interact with to display toggletips */
+  children: JSX.Element;
+}
+
+export const Toggletip = React.memo(
+  ({
+    children,
+    theme = 'info',
+    placement = 'auto',
+    content,
+    title,
+    closeButton = true,
+    onClose,
+    footer,
+  }: ToggletipProps) => {
+    const styles = useStyles2(getStyles);
+    const style = styles[theme];
+    const contentRef = useRef(null);
+    const [controlledVisible, setControlledVisible] = React.useState(false);
+
+    useEffect(() => {
+      if (controlledVisible) {
+        const handleKeyDown = (enterKey: KeyboardEvent) => {
+          if (enterKey.key === 'Escape') {
+            setControlledVisible(false);
+            if (!!onClose) {
+              onClose();
+            }
+          }
+        };
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+          document.removeEventListener('keydown', handleKeyDown);
+        };
+      } else if (controlledVisible === false && !!onClose) {
+        onClose();
+      }
+      return;
+    }, [controlledVisible, onClose]);
+
+    const { getArrowProps, getTooltipProps, setTooltipRef, setTriggerRef, visible, update } = usePopperTooltip({
+      visible: controlledVisible,
+      placement: placement,
+      interactive: true,
+      offset: [0, 8],
+      trigger: 'click',
+      onVisibleChange: setControlledVisible,
+    });
+
+    return (
+      <>
+        {React.cloneElement(children, {
+          ref: setTriggerRef,
+          tabIndex: 0,
+        })}
+        {visible && (
+          <Portal>
+            <div
+              data-testid="toggletip-content"
+              ref={setTooltipRef}
+              {...getTooltipProps({ className: style.container })}
+            >
+              {Boolean(title) && <div className={style.header}>{title}</div>}
+              {closeButton && (
+                <div data-testid="toggletip-header-close" className={style.headerClose}>
+                  <IconButton
+                    aria-label="Close Toggletip"
+                    name="times"
+                    size="md"
+                    onClick={() => setControlledVisible(false)}
+                  />
+                </div>
+              )}
+              <div ref={contentRef} {...getArrowProps({ className: style.arrow })} />
+              <div className={style.body}>
+                {(typeof content === 'string' || React.isValidElement(content)) && content}
+                {typeof content === 'function' && update && content({ update })}
+              </div>
+              {Boolean(footer) && <div className={style.footer}>{footer}</div>}
+            </div>
+          </Portal>
+        )}
+      </>
+    );
+  }
+);
+
+Toggletip.displayName = 'Toggletip';
+
+function getStyles(theme: GrafanaTheme2) {
+  function buildToggletipTheme(toggletipBg: string, toggletipBorder: string, toggletipText: string) {
```

Comment:
It makes me really skeptical that this is almost exactly the same as the tooltip one 😬 It will be so hard to keep them up to speed if we make changes in one of them... Can we put this function in some sort of helper and then use it in both tooltip and toggletip, I think a lot of the changes I can see (the negative margins on the arrows) we can try to make the same, its just the content padding that should really be changing and we can add a function parameter for that.
Do you think this is possible?